portal/account/index_reset.php

```diff
@@ -0,0 +1,170 @@
+    " FROM " . TBL_PAT_ACC_ON . " WHERE pid = ?";
+
+$auth = privQuery($sql, array($_SESSION['pid']));
+$code_current = trim($_POST['pass_current']);
+$current_encoded = oemr_password_hash($code_current, $auth[COL_POR_SALT]);
+if (isset($_POST['pass_current']) && $current_encoded != $auth[COL_POR_PWD]) {
+    $errmsg = xlt("Invalid Current Credentials Error.") . xlt("Current password.");
+    $logit->portalLog('Credential update attempt', '', ($_POST['uname'] . ':invalid password'), '', '0');
+    die($errmsg);
+}
+?>
+<!DOCTYPE html>
+<html>
+<head>
+    <title><?php echo xlt('Change Portal Credentials'); ?></title>
+    <?php
+    Header::setupHeader(['opener']);
+    ?>
+    <script type="text/javascript">
+        function process_new_pass() {
+            if (document.getElementById('login_uname').value != document.getElementById('confirm_uname').value) {
+                alert(<?php echo xlj('The Username fields are not the same.'); ?>);
+                return false;
+            }
+            if (document.getElementById('pass_new').value != document.getElementById('pass_new_confirm').value) {
+                alert(<?php echo xlj('The new password fields are not the same.'); ?>);
+                return false;
+            }
+            if (document.getElementById('pass_new').value == document.getElementById('pass_new_confirm').value) {
+                if (!confirm(<?php echo xlj('The new password is the same as the current password. Click Okay to accept anyway.'); ?>)) {
+                    return false;
+                }
+            }
+        }
+    </script>
+    <style>
+        .table > tbody > tr > td {
+            border-top: 0px;
+        }
+    </style>
+</head>
+<body
+phpmailer>
+    <br><br>
+    <div class="container">
+        <?php if (empty($_POST['submit'])) { ?>
+            <form action="" method="POST" onsubmit="return process_new_pass()">
+                <input style="display:none" type="text" name="dummyuname" />
+                <input style="display:none" type="password" name="dummypassword" />
+                <table class="table table-condensed" style="border-bottom:0px;width:100%">
+                    <tr>
+                        <td width="35%"><strong><?php echo xlt('Account Name'); ?><strong></td>
+                        <td><input class="form-control" name="uname" id="uname" type="text" readonly
+                                value="<?php echo attr($auth['portal_username']); ?>" /></td>
+                    </tr>
+                    <tr>
+                        <td><strong><?php echo xlt('New or Current Username'); ?><strong></td>
+                        <td><input class="form-control" name="login_uname" id="login_uname" type="text" required
+                                title="<?php echo xla('Change or keep current. Enter 8 to 20 characters'); ?>" pattern=".{8,20}"
+                                value="<?php echo attr($auth['portal_login_username']); ?>" />
```

Comment:
another thing for me to figure out:
What happens if the above $one_time token hash is stored here?

Reply:
It's not. Logic clears before making to UI. Even if it did make it there, validation would catch.

Reply:
Thanks for the explanation.